backend/app/services/chat/trigger/core.py

```
@@ -362,10 +365,58 @@ async def _stream_chat_response(
         span_manager.set_model_attributes(chat_config.model_config)
 
+        # Search for relevant memories (with timeout, graceful degradation)
+        # WebSocket chat (web) respects user preference for memory
+        # This runs before context processing to inject memory context
+        from app.services.memory import get_memory_manager, is_memory_enabled_for_user
+
+        memory_manager = get_memory_manager()
+        relevant_memories = []
+
+        # Check user preference first
+        if not is_memory_enabled_for_user(user):
```

Ruff (0.14.13)

Undefined name `user`

(F821)